components/odh-notebook-controller/controllers/notebook_dspa_secret.go

```diff
@@ -0,0 +1,372 @@
+/*
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package controllers
+
+import (
+	"context"
+	"encoding/json"
+	"fmt"
+	"reflect"
+
+	"github.com/go-logr/logr"
+	nbv1 "github.com/kubeflow/kubeflow/components/notebook-controller/api/v1"
+	dspav1 "github.com/opendatahub-io/data-science-pipelines-operator/api/v1"
+	corev1 "k8s.io/api/core/v1"
+	apierrs "k8s.io/apimachinery/pkg/api/errors"
+	"k8s.io/apimachinery/pkg/api/meta"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+
+	"k8s.io/apimachinery/pkg/runtime/schema"
+	"k8s.io/apimachinery/pkg/types"
+	"k8s.io/client-go/dynamic"
+	"k8s.io/client-go/rest"
+	"k8s.io/utils/ptr"
+	"sigs.k8s.io/controller-runtime/pkg/client"
+)
+
+const (
+	elyraRuntimeSecretName = "ds-pipeline-config"
+	elyraRuntimeMountPath  = "/opt/app-root/runtimes"
+	elyraRuntimeVolumeName = "elyra-dsp-details"
+	dashboardInstanceName  = "default-dashboard"
+	dspaInstanceName       = "dspa"
+	managedByKey           = "opendatahub.io/managed-by"
+	managedByValue         = "workbenches"
+)
+
+func getDashboardInstance(ctx context.Context, dynamicClient dynamic.Interface, log logr.Logger) (map[string]interface{}, error) {
+	dashboardGVR := schema.GroupVersionResource{
+		Group:    "components.platform.opendatahub.io",
+		Version:  "v1alpha1",
+		Resource: "dashboards",
+	}
+
+	obj, err := dynamicClient.Resource(dashboardGVR).Get(ctx, dashboardInstanceName, metav1.GetOptions{})
+	if err != nil {
+		if apierrs.IsNotFound(err) {
+			// Skipping as Dashboard CR not found (optional cr)
+			return nil, nil
+		}
+		log.Error(err, "Failed to retrieve Dashboard CR", "name", dashboardInstanceName)
+		return nil, err
```

Comment:
Should this also handle the `if meta.IsNoMatchError(err) {` case?

Reply:
done